### PR TITLE
MEN-1877: Add documentation for GRUB and x86 support.

### PR DESCRIPTION
**Pull request by kacf**
Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

File: 03.Devices/02.Partition-layout/docs.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ When [building a Mender Yocto Project image](../../artifacts/building-mender-yoc
...
 
-The standard Yocto Project `IMAGE_FSTYPES` variable will be used to determine the image types to create in Yocto deploy directory. The meta-mender layer will add the `.sdimg` and `.mender` types to that variable. The filesystem used for the individual partitions in the `.sdimg` and `.mender` files will be based on the `ARTIFACTIMG_FSTYPE` variable.  It is advised that you clean up the `IMAGE_FSTYPES` variable to avoid creating unnecessary image files.
+The standard Yocto Project `IMAGE_FSTYPES` variable will be used to determine the image types to create in Yocto deploy directory. The meta-mender layer will add the `mender` type to that variable, and usually either `sdimg` or `uefiimg`, depending on whether [UEFI layout is used or not](../../artifacts/image-configuration/features#list-of-features). The filesystem used for the individual partitions in the `sdimg`, `uefiimg` and `mender` files will be based on the `ARTIFACTIMG_FSTYPE` variable.  It is advised that you clean up the `IMAGE_FSTYPES` variable to avoid creating unnecessary image files.
```
Comment (pasinskim) [inline 03.Devices/02.Partition-layout/docs.md]: In the previous version, we've been referring to `mender` and `sdimg` variables as `.mender` and `.sdimg`.

File: 03.Devices/03.Integrating-with-U-Boot/02.Integration-checklist/docs.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ taxonomy:
 
 In order to ensure that the necessary components for Mender have been integrated correctly, use this checklist to verify each of them in turn. Use this when all components have been built successfully, and the device has booted correctly.
 
+!!! This checklist only applies if you are using U-Boot. If you are using another bootloader, such as GRUB, this section can be skipped.
```
Comment (pasinskim) [inline 03.Devices/03.Integrating-with-U-Boot/02.Integration-checklist/docs.md]: `This checklist applies only` sounds better.

File: 03.Devices/03.Integrating-with-U-Boot/docs.md [hunk 3]
```diff
@@ -68,7 +68,7 @@ Project you need to enable the `mender-uboot` feature using
 MENDER_FEATURES_ENABLE_append = " mender-uboot"
 ```
 
-!!! Note that if you have inherited the `mender-full` or `mender-full-ubi` class in your `local.conf`, then the `mender-uboot` feature is already on by default. See [the documentation on features](../../artifacts/image-configuration/features) for more information.
+!!! If the architecture is ARM, and the `mender-full` or `mender-full-ubi` class is inherited, then the `mender-uboot` feature is already on by default. See [the documentation on features](../../artifacts/image-configuration/features) for more information.
```
Comment (pasinskim) [inline 03.Devices/03.Integrating-with-U-Boot/docs.md]: Maybe we could leave `... is inherited in your local.conf, then ...`

File: 03.Devices/04.Integrating-with-GRUB/docs.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+This section deals with the Yocto configuration needed to integrate with the [GRUB bootloader](https://www.gnu.org/software/grub/?target=_blank).
+
+# Requirements
+
+In order to integrate with GRUB, your system must boot using the UEFI standard. The traditional BIOS based boot procedure is not supported.
+
+# Enabling GRUB
+
+To enable Mender integration with GRUB, the `mender-grub` feature should be turned on using `MENDER_FEATURES_ENABLE`. For instance, in your `local.conf`:
+
```
Comment (pasinskim) [inline 03.Devices/04.Integrating-with-GRUB/docs.md]: Maybe `You can enable it by placing the following line in your local.conf file:`?

File: 03.Devices/04.Integrating-with-GRUB/docs.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+---
+title: Integrating with GRUB
+taxonomy:
+    category: docs
+---
+
+This section deals with the Yocto configuration needed to integrate with the [GRUB bootloader](https://www.gnu.org/software/grub/?target=_blank).
```
Comment (pasinskim) [inline 03.Devices/04.Integrating-with-GRUB/docs.md]: How about `This section describes the Yocto configuration needed ...`?

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md [hunk 6]
```diff
@@ -6,7 +6,7 @@ taxonomy:
 
 This document outlines the steps needed to build a [Yocto Project](https://www.yoctoproject.org/?target=_blank) image for a device.
 The build output will most notably include:
-* a file that can be flashed to the device storage during initial provisioning, it has suffix `.sdimg`
+* a file that can be flashed to the device storage during initial provisioning. It has the suffix `.uefiimg` if using a UEFI partition layout (usually together with GRUB), and a `.sdimg` suffix if not (usually together with U-Boot).
```
Comment (pasinskim) [inline 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: I would reword to `It contains the `.uefimg` suffix if UEFI partition layout is used (usually together with GRUB), and a `.sdimg` suffix if not (usually together with U-Boot).`

File: 03.Devices/03.Integrating-with-U-Boot/docs.md (hunk 3 above)
Comment (kacf) [reply 03.Devices/03.Integrating-with-U-Boot/docs.md]: Yeah, I'm unsure. I removed it because strictly speaking, it doesn't have to be added `local.conf` and many people will probably add it in a more permanent place, such as their machine definition. I'll do the compromise, and write `... is inherited in a Bitbake .conf file`.

**Comment by kacf** [comment]
Everything else fixed! Thanks!
